src/DotNetLightning.Core/Channel/Channel.fs

```diff
@@ -44,7 +44,7 @@ module Channel =
 
     let private hex = NBitcoin.DataEncoders.HexEncoder()
     let private ascii = System.Text.ASCIIEncoding.ASCII
-    let private dummyPrivKey = Key(hex.DecodeData("0101010101010101010101010101010101010101010101010101010101010101"))
+    let private dummyPrivKey = new Key(hex.DecodeData("0101010101010101010101010101010101010101010101010101010101010101"))
```

Comment:
if Key is not disposable, you don't need the `new` keyword

Reply:
It has become disposable after Nicolas prepared `NBitcoin.Secp256k1`

Reply:
oh I see; I guess because of the interop with unmanaged code?

Reply:
There is no unmanaged code in NBitcoin, do you mean `NBitcoin.Secp256k1`, then yes.
It is because inner ECPrivKey object in NBitcoin.Secp256k1 is disposable.
I don't know why exactly this had to be disposable. Probably because in original secp256k1 it was clearing memory after using privkey to prevent sensitive infomation leaking in memory.

Reply:
> do you mean NBitcoin.Secp256k1, then yes.

Yeah I mean NBitcoin.Secp256k1. Mmm, then I wonder if we should create a version of NBitcoin.Secp256k1 that uses BouncyCastle instead of unmanaged interop, what do you think?

Reply:
> then I wonder if we should create a version of NBitcoin.Secp256k1 that uses BouncyCastle instead of unmanaged interop, what do you think?

I didn't get what you mean. I think you are misunderstanding that `NBitcoin.Secp256k1` is Native bindings to unmanaged code?  It is not. It is a pure C# port of the original [secp256k1](https://github.com/bitcoin-core/secp256k1).

Reply:
good to know, but then I don't understand why it's disposable now

Reply:
Because  `NBitcoin` depends on `NBitcoin.Secp256k1` and `NBitcoin.Key` holds `NBitcoin.Secp256k1.ECPrivKey` object. And this `ECPrivKey` is `IDisposable` so it makes sense to dispose it when the wrapping `Key` is disposed.

So why this ECPrivKey should be disposable? The reason I've mentioned above is my guess.

> I don't know why exactly this had to be disposable. Probably because in original secp256k1 it was clearing memory after using privkey to prevent sensitive infomation leaking in memory.

Nicolas wanted to make `NBitcoin.Secp256k1` to behave as much as the same with original secp256k1, He imitated the behavior of its finalizer with `Dispose` method.